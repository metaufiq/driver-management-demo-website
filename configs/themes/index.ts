const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };