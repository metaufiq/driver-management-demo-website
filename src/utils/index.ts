/**
 * formator number to two digits number with string type
 * @param {number} num number
 * @returns {string} two digits number with string type
 */
export const _padTo2Digits = (num:number): string => {
  return num.toString().padStart(2, '0');
}

/**
 * format ISO date to format DD-MM-YYYY
 * @param {string} ISODate ISO date
 * @returns {string} date with format DD-MM-YYYY
 */
export const formatISODateToDate = (ISODate: string) =>{
  const date = new Date(ISODate);
  
  return [
    _padTo2Digits(date.getDate()),
    _padTo2Digits(date.getMonth() + 1),
    date.getFullYear(),
  ].join('-');
}