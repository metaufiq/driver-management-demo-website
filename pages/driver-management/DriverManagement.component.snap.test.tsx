import { assertSnapshots } from "../../src/test";
import { Configs } from "../../src/test/index.types";
import { themes } from "../../src/configs/themes";
import DriverManagement from "./index.page";

jest.mock('../../src/hooks/useUsers', ()=>()=>({
  users: [{key: 'example one of users key'}]
}))
jest.mock('../../src/hooks/useUsersQuery', ()=>()=>({
  users: [{key: 'example one of users api key'}]
}))

describe('DriverCardSnapshot', () => {

  const configs: Configs<any> = [
    {
      desc: 'should render normal driver management page',
      theme: themes,
      useTheme: true
    }
  ];

  assertSnapshots(DriverManagement, configs);
});
