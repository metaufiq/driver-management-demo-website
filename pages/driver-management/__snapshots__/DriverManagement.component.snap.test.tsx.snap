// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverCardSnapshot should render normal driver management page 1`] = `
<DocumentFragment>
  <div
    class="sc-dIouRR jXOnOq"
  >
    <div>
      <h1
        class="sc-hHLeRK cXweQi"
      >
        DRIVER MANAGEMENT
      </h1>
      <p
        class="sc-dmRaPn dUVFjh"
      >
        Data Driver bekerja dengan Anda.
      </p>
    </div>
    <div
      class="sc-kgflAQ gLCdaI"
    >
      <div
        class="sc-gsnTZi hEqjwc"
      >
        <div
          class="sc-dkzDqf gZKilg"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </div>
        <input
          class="sc-bczRLJ gBziFm"
          placeholder="Cari Driver"
          type="text"
        />
      </div>
      <button
        class="sc-hKMtZM cbntAE"
      >
        TAMBAH DRIVER
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4v16m8-8H4"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>
  <br />
  <div
    class="sc-bjUoiL fghqXP"
  >
    <div
      class="sc-eCYdqJ iywVtZ"
    >
      <div
        class="sc-gKXOVf bmManZ"
      >
        <div>
          <p
            class="sc-jSMfEi sc-iBkjds iqvdNZ hxEleX"
          >
            Driver ID
          </p>
          <p
            class="sc-ftvSup BYxNv"
          />
        </div>
        <div
          class="sc-papXJ hdbybu"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-iqcoie dOItqw"
      >
        <img
          class="sc-ksZaOG dfRWDh"
        />
        <div
          class="sc-crXcEl eEnHtW"
        >
          <div
            class="sc-evZas ddeZRc"
          >
            <p
              class="sc-jSMfEi sc-jqUVSM iqvdNZ fYuGPZ"
            >
              Nama Driver
            </p>
            <p
              class="sc-kDDrLX gQtTej"
            >
              , 
            </p>
          </div>
          <div
            class="sc-evZas ddeZRc"
          >
            <p
              class="sc-jSMfEi sc-jqUVSM iqvdNZ fYuGPZ"
            >
              Telepon
            </p>
            <p
              class="sc-kDDrLX gQtTej"
            />
          </div>
          <div
            class="sc-breuTD hYHGIL"
          >
            <div
              class="sc-evZas ddeZRc"
            >
              <p
                class="sc-jSMfEi sc-jqUVSM iqvdNZ fYuGPZ"
              >
                Email
              </p>
              <p
                class="sc-kDDrLX gQtTej"
              />
            </div>
            <div
              class="sc-evZas ddeZRc"
            >
              <p
                class="sc-jSMfEi sc-jqUVSM iqvdNZ fYuGPZ"
              >
                Tanggal Lahir
              </p>
              <p
                class="sc-kDDrLX gQtTej"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-idiyUo eqxcDt"
  >
    <button
      class="sc-hKMtZM kBHTJJ"
      disabled=""
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="15 18 9 12 15 6"
        />
      </svg>
      Prev Page
    </button>
    <button
      class="sc-hKMtZM kBHTJJ"
    >
      Next Page
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
